Sources/SwiftGenKit/Parsers/CoreData/CoreDataParser.swift

```diff
@@ -31,42 +39,40 @@ public enum CoreData {
       self.warningHandler = warningHandler
     }
 
-    public func parse(path: Path) throws {
-      if path.extension == "xcdatamodeld" {
+    public static let defaultFilter = ".*\\.xcdatamodeld?"
+
+    public func parse(path: Path, relativeTo parent: Path) throws {
+      if path.extension == Constants.modelBundleExtension {
         guard let modelPath = findCurrentModel(path: path) else {
           throw ParserError.invalidFile(path: path, reason: "Current version of Core Data model could not be found.")
         }
-        try parse(path: modelPath)
-      } else if path.extension == "xcdatamodel" {
+        try parse(path: modelPath, relativeTo: parent)
+      } else if path.extension == Constants.modelExtension {
         try addModel(path: path)
       } else {
-        let dirChildren = path.iterateChildren(options: [.skipsHiddenFiles, .skipsPackageDescendants])
-
-        for file in dirChildren where file.extension == "xcdatamodeld" || file.extension == "xcdatamodel" {
-          try parse(path: file)
-        }
+        throw ParserError.invalidFormat(reason: "Unknown file type for \(path)")
```

Comment:
What happens if the user overrides the defaultFilter with their own and their filter doesn't limit to only those two extensions, like use `UserModel.*`, and in the directory they scan they have `UserModel1.xcdatamodel`, `UserModel2.xcdatamodel`, but also `UserModel3.json` and `UserModel4.xcdatamodel`? I mean, just wanted to be sure that the `ParserError` only interrupts the parsing of an unsupported file like `UserModel3.json` but doesn't interrupt the whole scan of the rest of the directory and that `UserModel4.xcdatamodel` still gets a chance to be parsed :wink:

Reply:
Well, you're feeding invalid data into the **core data** parser, so if you want to override the default filter, maybe ensure you're only passing in valid data?

This is something that applies to all the parsers.

Reply:
Yeah, of course, but what I meant was, we just need to make sure that a failure when parsing one file doesn't stop the parsing of the rest of the files. If the config file in SwiftGen asks to parse (a directory containing) 5 files, and file 3 is corrupted or errors, we should print the error but still continue parsing files 4 and 5. I think I remember in a previous version of SwiftGen that this was an issue (SwiftGen stopped on the first error it encountered and didn't continue on other files), instead of reporting the error and just skipping the problematic file as intended but continuing on the other files… so just wanted to be sure that behavior was still working properly as expected.

Reply:
Right, so as discussed in Slack, we're going to leave the "fail early" system as it is now (unchanged from before this PR), and we'll take a better look at this in a separate issue. Refs #547.